0011-CIR-trust-distribution.md

```diff
@@ -0,0 +1,116 @@
+# Dapr Trust Distribution
+
+* Author(s): joshvanl
+* State: Draft
+* Updated: 2023-10-24
+
+## Overview
+
+This is a design proposal to implement a proper trust distribution process in Dapr.
+Trust distribution will be implemented in a seamless way without downtime.
+This will improve and unlock security related features.
+
+## Background
+
+### Motivation
+
+To support the creation of features:
+
+- Proper Certificate Authority (CA) rotation (without re-using the root's private key)
+- External CA sources such as cert-manager and cloud provider CAs etc.
+- Dapr multi-cluster and networking federation
+
+Related issues:
+- [Multicluster Kubernetes](https://github.com/dapr/dapr/issues/3460)
+- [[Proposal] Support third-party CA - Integrate Cert Manager with Dapr](https://github.com/dapr/dapr/issues/3968)
+- [[Proposal] Automatic root certificate rotation](https://github.com/dapr/dapr/issues/5958)
+
+### Goals
+
+- Implement an active trust distribution mechanism for Dapr in Kubernetes that responds to updates.
```

Comment:
It will work outside of Kubernetes in the sense that programs will pickup new CA bundles on file, however managing the propagation of the files to the correct locations on disk is definitely a task left to the reader. The UX on Kubernetes will be a lot more automated and "one touch" as the filesystem is virtualised and consistent using ConfigMaps. For this reason, I only highlighted this mode as a goal as it wouldn't be fair to say CA rotation was first class in standalone.

Reply:
Ok. Can you clarify this too please?

Reply:
Done